File: 1.two-sum/golang/src/two_sum.go
package solution

func TwoSum(nums []int, target int) []int {
	m := map[int]int{}

	for i, num := range nums {
		diff := target - num

		if _, ok := m[num]; ok {
			return []int{m[num], i}
		}

		m[diff] = i
	}

	return []int{}
}

File: 1.two-sum/java/src/main/java/company/app/TwoSum.java
package company.app;

import java.util.Map;
import java.util.HashMap;

public class TwoSum
{
    public int[] _do(int[] nums, int target)
    {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            int current = nums[i];
            int diff = target - current;

            if (map.containsKey(current)) {
                return new int[] {map.get(current), i};
            }

            map.put(diff, i);
        }

        return new int[] {};
    }
}
